### DISCUSSION TITLE
Old Frequently Asked Discussion

### DISCUSSION THREAD

**Discussion by gokadzev** [gokadzev/Musify#112] [General]
# Moved to #340

**Comment by xplshn** [comment]
There dos this app gen its music from? I can't find that in the FAQ, readme.md, or the official site. Is this using YouTube? Or what APIs exactly, yt-dlp? Also, does the app have Opus support? Is it really streaming(with or without buffer) or does it download the whole thing?

**Reply by gokadzev** [reply]
> There dos this app gen its music from? I can't find that in the FAQ, readme.md, or the official site. Is this using YouTube? Or what APIs exactly, yt-dlp? Also, does the app have Opus support? Is it really streaming(with or without buffer) or does it download the whole thing?

Hello, the app gets music from YouTube. The app gets only original m4a, all other formats are just extension changes. It streams music, not downloading whole music and then playing.

**Reply by xplshn** [reply]
Have you thought of replacing the YouTube API with Invidious, or better, Piped? So you can get the YouTube Music thumbnails, playlists and better audio formats support, and real streaming, are you using the newpipe lib? If so maybe its an easy switch because piped and invidious both are drop in replacements for the official YouTube API

**Reply by gokadzev** [reply]
> Have you thought of replacing the YouTube API with Invidious, or better, Piped? So you can get the YouTube Music thumbnails, playlists and better audio formats support, and real streaming, are you using the newpipe lib? If so maybe its an easy switch because piped and invidious both are drop in replacements for the official YouTube API

i was thinking of replacing the current one with newpipe but there's no active implementation of it in dart + this one works good so...

**Reply by xplshn** [reply]
If it works,  no need to change it

**Comment by abdim-k** [comment]
What is the backend of this app? it works like spotify and i impressed

**Comment by abdim-k** [comment]
Can You Upload your own music and user acces it?

**Reply by gokadzev** [reply]
> Can You Upload your own music and user acces it?

Yes, upload it on Youtube and it will appear here too.

**Reply by abdim-k** [reply]
is this legal? how you did that sorry but can you tell me how it works this is genius bro

**Reply by xplshn** [reply]
> is this legal? how you did that sorry but can you tell me how it works this is genius bro

Be it legal or not, has no commercial use, nor entity, its just code, if he hasn't cracked Google servers, broken any terms of services, its legal, and if he has(probably terms of service), Google can't take this down because its open source, its not worth it, since everyone can download, fork and contribute to the project. They'd probably bring more attention to the project if they sent a cease and desist.

**Comment by Davy49** [comment]
As I just installed this app on my android phone a short time ago I'm still discovering how it functions, so far it seems pretty good. I was rather disappointed though when I performed a search for classic rock and I didn't receive any results

**Reply by gokadzev** [reply]
> As I just installed this app on my android phone a short time ago I'm still discovering how it functions, so far it seems pretty good. I was rather disappointed though when I performed a search for classic rock and I didn't receive any results

Hello, do you mean searching in playlists or searching in the music section?

**Comment by Davy49** [comment]
> As I just installed this app on my android phone a short time ago I'm still discovering how it functions, so far it seems pretty good. I was rather disappointed though when I performed a search for classic rock and I didn't receive any results

> Hello, do you mean searching in playlists or searching in the music section?

Hello,
Thanks so much for your response, since I just started using the app I'm still learning all of the functionality of the app. As far as my search results are concerned I just typed in the words 'Classic Rock' and was disappointed that I didn't receive any results.

**Reply by gokadzev** [reply]
> Hello, Thanks so much for your response, since I just started using the app I'm still learning all of the functionality of the app. As far as my search results are concerned I just typed in the words 'Classic Rock' and was disappointed that I didn't receive any results. 

I will add more playlists including Classic rock playlists then. In general searching using genre in music search won't work, only artist or/and music title.

Also If you have favorite playlists on YouTube, you have the possibility to add yourself to the User Playlists section using the playlist ID.

**Comment by Davy49** [comment]
> Hello, Thanks so much for your response, since I just started using the app I'm still learning all of the functionality of the app. As far as my search results are concerned I just typed in the words 'Classic Rock' and was disappointed that I didn't receive any results.

>I will add more playlists including Classic rock playlists then. In general searching using genre in music search won't work, only artist or/and music title.

>Also If you have favorite playlists on YouTube, you have the possibility to add yourself to the User Playlists section using the playlist ID.


Hi,
I’m totally impressed that I actually received any kind of a response at all, I really do appreciate it !! I’m still in the learning stage of determing all of the functionality of the musify application so I view your reply as a positive one.
David W.

**Comment by bassmannate** [comment]


**Reply by Rexelpitlum** [reply]
Would it be a big issue to reintroduce an (perhaps optional) download folder selection?

My Music folder is located on the (limited) internal storage, whereas I would have practically limitless external storage space on the SD card (that I normally use for my music collections).

**Reply by bassmannate** [reply]
I see now. For some reason, it must have been failing to complete the downloads. Thanks!

**Reply by gokadzev** [reply]
> I see now. For some reason, it must have been failing to complete the downloads. Thanks!

It shouldn't fail since the correct folder is chosen for the correct extension.

**Reply by gokadzev** [reply]
> Would it be a big issue to reintroduce an (perhaps optional) download folder selection?
> 
> My Music folder is located on the (limited) internal storage, whereas I would have practically limitless external storage space on the SD card (that I normally use for my music collections).

@Rexelpitlum Hello, I apologize for not responding to your message earlier. I preferred the default music folder due to the new limitations introduced in Android 13. Users were choosing various folders, and this was causing downloads to fail. However, if you would like to revert to the previous setup, I can try to find a solution for it.

**Comment by OmarAMG** [comment]
Hola tengo un problema quisiera agregar playlist de Youtube pero al poner el link me tira un error que dice algo sobre que no es un ID y no se en que forma podría agregarlas entonces, además tampoco puedo crearlas dentro de la misma app.

**Reply by mistery-7** [reply]
Estoy teniendo el mismo problema.

**Reply by gokadzev** [reply]
@OmarAMG Hello, you need playlist id, not playlist link. check this: https://github.com/gokadzev/Musify/discussions/112#discussioncomment-6945247

**Comment by mistery-7** [comment]
First of all, congratulations on developing this amazing app! But, I would like to know how to add playlists. Asks to enter a playlist ID. Where do I get these IDs?

**Reply by gokadzev** [reply]
> First of all, congratulations on developing this amazing app! But, I would like to know how to add playlists. Asks to enter a playlist ID. Where do I get these IDs?

Hello, when you copy any playlist link, it looks like this (this is example link, after list= code will be different for different playlists): https://www.youtube.com/playlist?list=PLcirGkCPmbmFeQ1sm4wFciF03D_EroIfr


the only thing you need is remove everything except the code after list=, so as for this example, the playlist id will be: PLcirGkCPmbmFeQ1sm4wFciF03D_EroIfr

~~https://www.youtube.com/playlist?list=~~ PLcirGkCPmbmFeQ1sm4wFciF03D_EroIfr

**Comment by SFPunishment** [comment]
En la aplicación de Youtube Music tu puedes limitar la calidad de las canciones para que tus datos no se gasten tan rápido. Hay alguna forma de hacer esto en Musify?

**Comment by codeambulance** [comment]
I think the app will get perfect.
But since the last update i can not play a youtube playlist.
I only see grey squares. Moto g8 plus Android 10.

![musify_playlist_problem.png](https://github.com/gokadzev/Musify/assets/140640429/6f19f97a-c4c2-4acc-9a3b-4a8ff6722096)

**Reply by gokadzev** [reply]
@codeambulance Will be fixed in the next release :)

**Comment by leonardoquevedox** [comment]
Hi, @gokadzev! First of all, thanks for putting the effort on developing Musify: I was in need of an easy way to play YouTube music in background and Musify worked really great for me! 💛

I was wondering: Are you planning on adding iOS support for it? Since the app is built in Flutter I think it would be awesome to take advantage of its cross platform powers ⚡

**Reply by gokadzev** [reply]
> Hi, @gokadzev! First of all, thanks for putting the effort on developing Musify: I was in need of an easy way to play YouTube music in background and Musify worked really great for me! 💛
> 
> I was wondering: Are you planning on adding iOS support for it? Since the app is built in Flutter I think it would be awesome to take advantage of its cross platform powers ⚡

Hello @leonardoquevedox, I'm glad you liked the app. 


Besides the fact that I can't test it since I don't own a Mac, not having an Apple Developer account is another issue (if I'm not mistaken, its annual fee is $99). That's why I primarily develop this app for Android.

**Comment by S4r4h-O** [comment]
Hi.

Your app is very good, but I cannot add a playlist. I want to add https://youtube.com/playlist?list=PLvVBIN_eUn9YsvnM1DWR8ahpKm3mRqJSW&si=dc2QP_hcpRO5FbI4, and I know the ID is PLvVBIN_eUn9YsvnM1DWR8ahpKm3mRqJSW&si=dc2QP_hcpRO5FbI4, but the app displays the error "This is not a playlist ID":

![Screenshot_20231207-132146](https://github.com/gokadzev/Musify/assets/128503216/0aa67afd-ac16-40ed-bb49-fc10925b9352)

**Reply by S4r4h-O** [reply]
I figured it out, the ID is just PLvVBIN_eUn9YsvnM1DWR8ahpKm3mRqJSW. Nice app ;)

**Reply by gokadzev** [reply]
> I figured it out, the ID is just PLvVBIN_eUn9YsvnM1DWR8ahpKm3mRqJSW. Nice app ;)

Thanks :)

**Comment by craniumsnumb** [comment]
I added this here and the issues channel to make sure it got seen. It won't let me save my user data for some reason. It displays an error code shown in the screenshot I will attach.![Screenshot_20231207-183700.png](https://github.com/gokadzev/Musify/assets/153243999/cb68ae58-7ee7-45c5-b9f4-19fde747a023)
